### PR TITLE
Upgrade to recrypt 0.12.0

### PR DESCRIPTION
**Pull request by giarc3**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -33,7 +33,7 @@ jobs:
             features: --features tls-vendored-openssl --no-default-features
           # MSRV
           - os: ubuntu-18.04
-            toolchain: 1.39.0
+            toolchain: 1.41.1
```
Comment (clintfred) [inline .github/workflows/ci.yaml:36]: Matches recrypt's MSRV.

**Review by clintfred — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,9 @@
   - Update to reqwest 0.11.0
   - Update to bytes 1.0
   - Require minimum protobuf of 2.20.0
+- [[#215](https://github.com/IronCoreLabs/ironoxide/pull/215)]
+  - Update to recrypt 0.12.0
+  - Bump MSRV to 1.41.1
```
Comment (BobWall23) [inline CHANGELOG.md:12]: Isn't this only pertinent to our build process? I think it might be best to omit that from the changelog.

**Review by BobWall23 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (clintfred) [reply CHANGELOG.md:12]: MSRV applies to anyone depending on ironoxide. In other words, if you want to depend on ironoxide, you must be using rustc 1.41.1 or higher.

Comment (giarc3) [reply CHANGELOG.md:12]: Are you referring to the MSRV? That's the minimum version of rust that we say we support (and test against in CI), so I think it's important to include that for others to know.

Comment (BobWall23) [reply CHANGELOG.md:12]: OK - agreed.

**Review by BobWall23 — APPROVED** [review]

**Pull request merged**